### PR TITLE
Python Converter and Json for the ADPUPA BUFR DUMP

### PR DESCRIPTION
**Pull request by PraveenKumar-NOAA**
This PR adds an ioda-converters Python script for the ADPUPA BUFR DUMP. The following two files are new:

~/ush/ioda/bufr2ioda/bufr2ioda_adpupa.py
~/parm/ioda/bufr2ioda/bufr2ioda_adpupa.json

Validation for the output obs is performed using the pressure and temperature observations directly from the input BUFR and output IODA file, please see #978. The end-to-end tests were also run. 

The additional informations for the drift latitude/longitude and for the derived variables like specific humidity will be added separately.

### CODE REVIEW

File: ush/ioda/bufr2ioda/bufr2ioda_adpupa.py [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+    # LaunchTime
+    obsspace.create_var('MetaData/launchTime', dtype=timestamp.dtype, fillval=timestamp.fill_value) \
+        .write_attr('units', 'seconds since 1970-01-01T00:00:00Z') \
+        .write_attr('long_name', 'Launch Time') \
+        .write_data(launchtime)
+
+    # Air Temperature
+    obsspace.create_var('ObsValue/airTemperature', dtype=tmdb.dtype, fillval=tmdb.fill_value) \
+        .write_attr('units', 'K') \
+        .write_attr('long_name', 'Sensible Temperature') \
```
Comment (nicholasesposito) [inline ush/ioda/bufr2ioda/bufr2ioda_adpupa.py]: I wonder if this should be "Sensible Air Temperature" ?

**Review by nicholasesposito — DISMISSED** [review]
Looks good to me.

File: ush/ioda/bufr2ioda/bufr2ioda_adpupa.py (hunk 1 above)
Comment (PraveenKumar-NOAA) [reply ush/ioda/bufr2ioda/bufr2ioda_adpupa.py]: @nicholasesposito thanks, replaced the Sensible Temperature by 'Air Temperature'.

**Review by nicholasesposito — APPROVED** [review]
Approved. Thanks, Praveen.

**Review by ADCollard — APPROVED** [review]

**Pull request merged**